src/common/utils/oidc/helper.go

```diff
@@ -362,6 +375,29 @@ func GroupsFromClaims(gp claimsProvider, k string) ([]string, bool) {
 	return res, true
 }
 
+type populate func(groupNames []string) ([]int, error)
+
+func populateGroupsDB(groupNames []string) ([]int, error) {
+	return group.PopulateGroup(models.UserGroupsFromName(groupNames, common.OIDCGroupType))
+}
+
+// InjectGroupsToUser populates the group to DB and inject the group IDs to user model.
+// The third optional parm is for UT only, when using the func, the third
+func InjectGroupsToUser(info *UserInfo, user *models.User, f ...populate) {
```

Comment:
is it necessary to add the nil checking for info and user?

Reply:
Yep, it's not likely to happen but I'll add the check thanks.